lib/jxl/modular/transform/squeeze.h

```diff
@@ -104,28 +103,28 @@ void InvHSqueeze(Image &input, int c, int rc, ThreadPool *pool) {
         pixel_type *JXL_RESTRICT p_out = chout.Row(y);
 
         // special case for x=0 so we don't have to check x>0
-        pixel_type_w avg = p_avg[0];
-        pixel_type_w next_avg = (1 < chin.w ? p_avg[1] : avg);
-        pixel_type_w tendency = SmoothTendency(avg, avg, next_avg);
-        pixel_type_w diff = p_residual[0] + tendency;
-        pixel_type_w A =
+        pixel_type avg = p_avg[0];
+        pixel_type next_avg = (1 < chin.w ? p_avg[1] : avg);
+        pixel_type tendency = SmoothTendency(avg, avg, next_avg);
+        pixel_type diff = p_residual[0] + tendency;
```

Comment:
this could overflow in theory, no? We probably should cast one of the two to a wider type...

Reply:
(also below)

Reply:
The calculation in SmoothTendency() could certainly overflow for >29-bit values, and probably things here too.

With > 24-bit integers we have a problem anyway though (converting them to float is lossy), so I'm not sure if it really makes sense to prevent overflow here if we can't really handle it anyway. On older 32-bit devices the widening probably does have a more significant cost...

Reply:
Well, signed integer overflow is UB and sanitizers/fuzzers complain about it, so doing something else there would be useful... perhaps just casting to u32, doing the arithmetic and then going back to i32 is enough.

Reply:
Doing that for RCT, reverting to wider arithmetic in Squeeze...